src/tribler-core/tribler_core/modules/resource_monitor/core.py

```diff
@@ -0,0 +1,105 @@
+import os
+import time
+
+from ipv8.taskmanager import TaskManager
+
+import psutil
+
+from tribler_common.simpledefs import NTFY
+
+from tribler_core.modules.resource_monitor.base import ResourceMonitor
+
+FREE_DISK_THRESHOLD = 100 * (1024 * 1024)  # 100MB
+DEFAULT_RESOURCE_FILENAME = "resources.log"
+
+
+class CoreResourceMonitor(ResourceMonitor, TaskManager):
+    """
+    Implementation class of ResourceMonitor by the core process. The core process uses
+    TaskManager to implement start() and stop() methods.
+    """
+
+    def __init__(self, session):
+        TaskManager.__init__(self)
+        ResourceMonitor.__init__(self)
+        self.session = session
+
+        self.disk_usage_data = []
+
+        self.state_dir = session.config.get_state_dir()
+        self.resource_log_file = session.config.get_log_dir() / DEFAULT_RESOURCE_FILENAME
+        self.resource_log_enabled = session.config.get_resource_monitor_enabled()
+
+    def start(self):
+        """
+        Start the resource monitor by scheduling a LoopingCall.
+        """
+        self.register_task("check_resources", self.check_resources,
+                           interval=self.session.config.get_resource_monitor_poll_interval())
+
+    async def stop(self):
```

Comment:
**question:** not sure is it correct or not -- to override non-async method by async method

Reply:
This is a good point, I'm not sure, seems to work though. I now made them the responsibility of the derived classes, removing from the base class.